src/controllers/action-queue/ActionQueueStore.js

```diff
@@ -95,8 +95,7 @@ export const makeActionQueueStore = (account: EdgeAccount): ActionQueueStore =>
   return instance
 }
 
-// TODO: Sam's recommendation: Create separate map in redux to manage running borrow action programs?
-export const useRunningActionQueueId = (borrowActionId: BorrowActionId) => {
+export const useRunningActionQueueId = (borrowActionId: BorrowActionId, walletId: string) => {
   const actionQueue: ActionQueueMap = useSelector(state => state.actionQueue.queue)
-  return Object.keys(actionQueue).find(programId => programId.includes('loan-create'))
+  return Object.keys(actionQueue).find(programId => programId.includes(borrowActionId) && programId.includes(walletId))
```

Comment:
No, it's a valid point. You brought up the notion of setting up an interface in the gui to map the data in a way that makes sense to the gui, so that it is isolated. I don't think this creates much technical debt for now, though.

Reply:
To be updated after LoanManager implementation.

Reply:
Yes, we'll refactor this in the LoanManager work.